### PR TITLE
Captcha as required element

### PR DESCRIPTION
**Pull request by ronen-erez**
This change enables users to use the captcha as required element in a form.
Thanks

### CODE REVIEW

File: packages/wix-ui-core/src/components/captcha/Captcha.driver.ts [hunk 1]
```diff
@@ -5,10 +5,10 @@ import {
 import {UniDriver} from 'unidriver';
 
 export interface CaptchaDriver extends BaseUniDriver {
-  /** clicks on the captcha checkbox challenge */
   getTheme: () => Promise<string>;
   getSize: () => Promise<string>;
   getLang: () => Promise<string>;
+  getRequired: () => Promise<string>;
```
Comment (shlomitc) [inline packages/wix-ui-core/src/components/captcha/Captcha.driver.ts]: `isRequired`, make sure it returns a boolean value

File: packages/wix-ui-core/src/components/captcha/Captcha.st.css [hunk 2]
```diff
@@ -21,3 +23,10 @@
 .root .captcha {
     display: none;
 }
+/* a hidden check box that will give the captcha a required input element capabilities */
+.root .checkbox {
```
Comment (shlomitc) [inline packages/wix-ui-core/src/components/captcha/Captcha.st.css]: I would call it `requiredField`

File: packages/wix-ui-core/src/components/captcha/test-assets/CaptchaTestComponent.testDriver.ts [hunk 3]
```diff
@@ -14,6 +14,11 @@ async function isCaptchaVerified() {
   return verifiedToken !== constants.verifiedTokenMark && verifiedToken.includes(constants.verifiedTokenMark);
 }
 
+async function isCaptchaRequired() {
+  const input = await $('input');
```
Comment (shlomitc) [inline packages/wix-ui-core/src/components/captcha/test-assets/CaptchaTestComponent.testDriver.ts]: why not use the same driver strategy for it? reuse the component driver that checks if the captcha is required

File: packages/wix-ui-core/src/components/captcha/test-assets/CaptchaTestComponent.tsx [hunk 4]
```diff
@@ -36,33 +36,46 @@ export class CaptchaTestComponent extends React.Component {
...
+            <div data-hook={constants.resetDataHook}>{constants.resetMark}</div>
+          )}
+          <button
+            onClick={this.resetCaptcha}
+            type="button"
+            data-hook={constants.resetButtonDataHook}
+          >
+            Click to reset captcha
+          </button>
+          <button type="submit" value="Submit Captcha">
```
Comment (shlomitc) [inline packages/wix-ui-core/src/components/captcha/test-assets/CaptchaTestComponent.tsx]: I don't see this is in use in our tests, what is it for?

File: packages/wix-ui-core/src/components/captcha/Captcha.st.css [hunk 5]
```diff
@@ -21,3 +23,10 @@
...
+    position: absolute;
+    opacity : 0.0;
+    bottom: 0;
```
Comment (shlomitc) [inline packages/wix-ui-core/src/components/captcha/Captcha.st.css]: is this the right placement for it? shouldn't it be in the center?
this is how it looks like now
image

File: packages/wix-ui-core/src/components/captcha/test-assets/CaptchaTestComponent.tsx (hunk 4 above)
Comment (ronen-erez) [reply packages/wix-ui-core/src/components/captcha/test-assets/CaptchaTestComponent.tsx]: if you want to manually test the required field than you can submit and see the results

Comment (shlomitc) [reply packages/wix-ui-core/src/components/captcha/test-assets/CaptchaTestComponent.tsx]: so let's create another example component that just displays another captcha with this button, to separate it from the test component.
you can also remove the added e2e test (its covered in the JSDOM tests)

File: packages/wix-ui-core/src/components/captcha/Captcha.st.css (hunk 5 above)
Comment (ronen-erez) [reply packages/wix-ui-core/src/components/captcha/Captcha.st.css]: i know.
is there a way to get the captcha position so i can relate the input to it? if we put center (what i have tried) it comes out in the center of the div ... out of the captcha element borders

File: packages/wix-ui-core/src/components/captcha/Captcha.st.css (hunk 2 above)
Comment (ronen-erez) [reply packages/wix-ui-core/src/components/captcha/Captcha.st.css]: done

File: packages/wix-ui-core/src/components/captcha/Captcha.driver.ts (hunk 1 above)
Comment (ronen-erez) [reply packages/wix-ui-core/src/components/captcha/Captcha.driver.ts]: done

File: packages/wix-ui-core/src/components/captcha/test-assets/CaptchaTestComponent.testDriver.ts (hunk 3 above)
Comment (ronen-erez) [reply packages/wix-ui-core/src/components/captcha/test-assets/CaptchaTestComponent.testDriver.ts]: can you pls elaborate? 
I assumed that required input is a tested feature and i verified we have an input.

**Pull request merged**